plugin/src/main/java/org/gradle/testretry/internal/executer/RetryTestResultProcessor.java

```diff
@@ -124,8 +127,27 @@ public void output(Object testId, TestOutputEvent testOutputEvent) {
         delegate.output(testId, testOutputEvent);
     }
 
-    @Override
+    @SuppressWarnings("unused")
     public void failure(Object testId, Throwable throwable) {
+        // Gradle 7.6 changed the method signature from failure(Object, Throwable) to failure(Object, TestFailure).
+        // To maintain compatibility with older versions, the original method needs to exist and needs to call failure()
+        // on the delegate via reflection.
+        failure(testId);
+        try {
+            Method failureMethod = delegate.getClass().getMethod("failure", Object.class, Throwable.class);
+            failureMethod.invoke(delegate, testId, throwable);
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void failure(Object testId, TestFailure result) {
```

Comment:
❓ Won't this cause problems when loading this class on previous Gradle versions?

Reply:
According to our cross-version tests, this works: https://e.grdev.net/s/eoxryno34i4n6

Reply:
That was my assumption, the TestFailure class should be loaded just-in-time, before the `failure(Object, TestFailure)` method is called.